extensions/spring-xd-extension-process/src/main/java/org/springframework/xd/extension/process/ShellCommandProcessor.java

```diff
@@ -129,15 +129,18 @@ public void start() {
 	 */
 	public String receive() {
 		Assert.isTrue(isRunning(), "Shell process is not started.");
-		String data;
-		try {
-			byte[] buffer = serializer.deserialize(stdout);
-			data = new String(buffer, charset);
-		}
-		catch (IOException e) {
-			throw new RuntimeException(e.getMessage(), e);
+		synchronized (process) {
```

Comment:
Since each `ShellCommandProcessor` has one `process` and the `receive` & `send` are wrapped in a synchronized block (except for the `Assert`), does it make sense to simply mark the methods themselves as synchronized?